test/DSROracle.t.sol

```diff
@@ -0,0 +1,198 @@
+        assertEq(oracle.getConversionRate(), 1e27);
+        assertEq(oracle.getConversionRate(block.timestamp + 365 days), 1e27);
+
+        pot.setDSR(DSR_FIVE_PCT_APY);
+        pot.setChi(1.03e27);
+        oracle.refresh();
+
+        assertEq(oracle.getConversionRate(), 1.03e27);
+        assertEq(oracle.getConversionRate(block.timestamp + 365 days), 1.081499999999999999959902249e27);
+    }
+
+    function test_gas_getConversionRate_1hour() public {
+        vm.pauseGasMetering();
+        pot.setDSR(DSR_FIVE_PCT_APY);
+        pot.setChi(1.03e27);
+        oracle.refresh();
+        vm.resumeGasMetering();
+
+        oracle.getConversionRate(block.timestamp + 1 hours);
+    }
+
+    function test_gas_getConversionRate_1year() public {
+        vm.pauseGasMetering();
+        pot.setDSR(DSR_FIVE_PCT_APY);
+        pot.setChi(1.03e27);
+        oracle.refresh();
+        vm.resumeGasMetering();
+
+        oracle.getConversionRate(block.timestamp + 365 days);
+    }
+
+    function test_getConversionRate_pastRevert() public {
+        vm.expectRevert();
+        oracle.getConversionRate(block.timestamp - 365 days);
+    }
+
+    function test_getConversionRateBinomialApprox() public {
+        assertEq(oracle.getConversionRateBinomialApprox(), 1e27);
+        assertEq(oracle.getConversionRateBinomialApprox(block.timestamp + 365 days), 1e27);
+
+        pot.setDSR(DSR_FIVE_PCT_APY);
+        pot.setChi(1.03e27);
+        oracle.refresh();
+
+        assertEq(oracle.getConversionRateBinomialApprox(), 1.03e27);
+        assertEq(oracle.getConversionRateBinomialApprox(block.timestamp + 365 days), 1.081495968383924399665215760e27);
+    }
+
+    function test_gas_getConversionRateBinomialApprox_1hour() public {
+        vm.pauseGasMetering();
+        pot.setDSR(DSR_FIVE_PCT_APY);
+        pot.setChi(1.03e27);
+        oracle.refresh();
+        vm.resumeGasMetering();
+
+        oracle.getConversionRateBinomialApprox(block.timestamp + 1 hours);
+    }
+
+    function test_gas_getConversionRateBinomialApprox_1year() public {
+        vm.pauseGasMetering();
+        pot.setDSR(DSR_FIVE_PCT_APY);
+        pot.setChi(1.03e27);
+        oracle.refresh();
+        vm.resumeGasMetering();
+
+        oracle.getConversionRateBinomialApprox(block.timestamp + 365 days);
+    }
+
+    function test_getConversionRateBinomialApprox_pastRevert() public {
+        vm.expectRevert();
+        oracle.getConversionRateBinomialApprox(block.timestamp - 365 days);
+    }
+
+    function test_getConversionRateLinearApprox() public {
+        assertEq(oracle.getConversionRateLinearApprox(), 1e27);
+        assertEq(oracle.getConversionRateLinearApprox(block.timestamp + 365 days), 1e27);
+
+        pot.setDSR(DSR_FIVE_PCT_APY);
+        pot.setChi(1.03e27);
+        oracle.refresh();
+
+        assertEq(oracle.getConversionRateLinearApprox(), 1.03e27);
+        assertEq(oracle.getConversionRateLinearApprox(block.timestamp + 365 days), 1.078790164207174267760128000e27);
```

Comment:
Similarly here, I'd be nice to have at least a little description to this value.

Reply:
done